hack/make-rules/test.sh

```diff
@@ -69,6 +69,10 @@ KUBE_COVER_REPORT_DIR="${KUBE_COVER_REPORT_DIR:-}"
 # coverage mode.
 KUBE_COVERPROCS=${KUBE_COVERPROCS:-4}
 KUBE_RACE=${KUBE_RACE:-}   # use KUBE_RACE="-race" to enable race testing
+# The idiomatic way to disable test caching explicitly is to use -count=1
```

Comment:
I would recommend instead that callers use `KUBE_RACE=-race` (which is a really awkward "api" but whatever) because `-race` also does not cache _and_ is what we run in CI _and_ will help surface flakes.

Reply:
Should we default to that?

Reply:
Maybe, it comes at the cost of any caching. I do think in an ideal world matching CI tests runs is as simple as `make test` (no args, no special funny business, just clone the branch and run the tests)

Reply:
cc @liggitt (we've discussed test runs, caching, etc. previously)